govcd/user_test.go

```diff
@@ -0,0 +1,219 @@
+// +build user functional ALL
+
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	. "gopkg.in/check.v1"
+)
+
+/*
+  TODO: Add test for takeOwnership.
+
+This is more complicated than it looks, because it requires the following:
+
+Either:
+1a. Separate connection with a newly created user [requires test enhancement]
+2a. Creation of entities with new user (vapp/catalog/catalog items)
+
+OR
+1b. create entities with the user that runs the tests
+2b. change ownership of such entities to the new user [requires new feature]
+
+3. Check that the user is the intended one (this is currently doable, because we can
+  inspect the Owner structure of the entity being created)
+
+4. Try deleting the user that owns the new entities
+5. get an error
+6. take ownership from the user
+7. delete the user and see the operation succeed
+8. Check that the new entities belong to the current user
+9. Delete the new entities
+*/
+
+// Checks that the default roles are available from the organization
+func (vcd *TestVCD) Test_GetRole(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip(fmt.Sprintf(TestRequiresSysAdminPrivileges, check.TestName()))
+	}
+	adminOrg, err := GetAdminOrgByName(vcd.client, vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, Not(Equals), AdminOrg{})
+	Roles := []string{
+		OrgUserRoleOrganizationAdministrator,
+		OrgUserRoleVappUser,
+		OrgUserRoleCatalogAuthor,
+		OrgUserRoleConsoleAccessOnly,
+	}
+	for _, roleName := range Roles {
+		// fmt.Printf("# retrieving role %s\n", roleName)
+		roleReference, err := adminOrg.GetRole(roleName)
+		check.Assert(err, IsNil)
+		check.Assert(roleReference, NotNil)
+		check.Assert(roleReference.Name, Equals, roleName)
+		check.Assert(roleReference.HREF, Not(Equals), "")
+	}
+}
+
+// Checks that we can retrieve an user by name or ID
```

Comment:
Grammar: "an user" -> "a user" (repeating in other places as well)

Reply:
done